src/tarball.js

```diff
@@ -94,6 +100,10 @@ export async function scanDirOrArchive(name, version, options) {
 
     ref.warnings.push(...fileAnalysisResults.flatMap((row) => row.warnings));
 
+    if (KVersionsToWarn.has(version)) {
```

Comment:
Maybe better to match with a RegExp like `^0[\.]?`.. Because the condition will not work for version like `0.1.0`